app/src/main/java/com/bignerdranch/android/geoquiz/MainActivity.kt

```diff
@@ -121,22 +122,68 @@ class MainActivity : AppCompatActivity() {
     private fun checkAnswer(userAnswer: Boolean) {
         val correctAnswer = quizViewModel.currentQuestionAnswer
 
+        trueButton.isEnabled = false
+        falseButton.isEnabled = false
+        quizViewModel.questionIsAnswered()
+
         val messageResId = when {
             quizViewModel.isCheater -> R.string.judgment_toast
             userAnswer == correctAnswer -> R.string.correct_toast
             else -> R.string.incorrect_toast
         }
 
-        val toast = Toast.makeText(this, messageResId, Toast.LENGTH_SHORT)
-        toast.setGravity(Gravity.TOP, 0, 300)
-        toast.show()
+        if (userAnswer == correctAnswer) {
+            pointForCorrectAnswer()
+        }
+
+        val toastAnswerStatus = Toast.makeText(
+            this,
+            messageResId,
+            Toast.LENGTH_SHORT
+        )
+        toastAnswerStatus.setGravity(Gravity.TOP, 0, 300)
+        toastAnswerStatus.show()
+
+        val scoreToast = Toast.makeText(
+                this,
+                "Twój wynik to " +
+                        "${(quizViewModel.score / quizViewModel.questionBankSize * 100).toInt()}%",
+                Toast.LENGTH_LONG
+            )
+        scoreToast.setGravity(Gravity.TOP, 0, 150)
+
+        if (quizViewModel.questionBankAreAnswered) {
+            toastAnswerStatus.show()
+            scoreToast.show()
+        } else {
+            toastAnswerStatus.show()
+        }
+    }
+
+    private fun updateButtonsEnabledState() {
+        val isQuestionAnswered = quizViewModel.isCurrentQuestionAnswered
+        trueButton.isEnabled = !isQuestionAnswered
+        falseButton.isEnabled = !isQuestionAnswered
+    }
+
+    private fun pointForCorrectAnswer() {
+        quizViewModel.score++
```

Comment:
normally, you would do this differently. You have learned about encapsulation already. If you have a component that exposes some data, its better to mutate this data only from within. On this example, that would mean to have a `private` method in the viewModel, i.e `incrementScore` and a public method `updateScore`. Then from here you would call `updateScore` only, but in the viewModel you would have:
```
fun updateScore() {
   this.incrementScore()
}

private fun incrementScore() {
   this.score++
}
```

This is important, as it defines the contract between components, meaning what you can and can't do. Having a public `updateScore` you do not directly mutate any state of it, but delegate it to it's internal functions. It also helps with testing later on

Reply:
Thanks, made changes as you suggested